wiring/src/spark_wiring_fuel.cpp

```diff
@@ -119,6 +119,9 @@ float FuelGauge::getSoC() {
 	byte LSB = 0;
 
 	readRegister(SOC_REGISTER, MSB, LSB);
+    if (MSB > 99 || LSB == 255) {
```

Comment:
Nit: formatting (spaces/tabs).

Also, this does not seem like a very reliable approach to track errors. If we are trying to fix this, `readRegister()` should be checking for errors and propagating them as well.

Reply:
That also makes sense. As `readRegister()` is private, we can change its prototype at our wish.

Reply:
We can also do this as a separate PR as the main thing we are trying to resolve here is an unrecoverable (without manual intervention) bus lockup after an error.